go/mysql/fakesqldb/server.go

```diff
@@ -414,6 +411,17 @@ func (db *DB) HandleQuery(c *mysql.Conn, query string, callback func(*sqltypes.R
 }
 
 func (db *DB) comQueryOrdered(query string) (*sqltypes.Result, error) {
+	var afterFn func() = func() {}
```

Comment:
There was only one other use of `OrderMatters()` in the code base. I refactored this to suit the purposes of this PR a bit better. The main change is that when `SetNeverFail(true)` is enabled, this path won't fail when there are non-matching queries, it will advance forward when there are matching queries.